src/mlpack/methods/lars/lars.cpp

```diff
@@ -68,6 +68,106 @@ LARS::LARS(const arma::mat& data,
   Train(data, responses, transposeData);
 }
 
+// Copy Constructor.
+LARS::LARS(const LARS& other) :
+    matGramInternal(other.matGramInternal),
+    matGram(&matGramInternal),
+    matUtriCholFactor(other.matUtriCholFactor),
+    useCholesky(other.useCholesky),
+    lasso(other.lasso),
+    lambda1(other.lambda1),
+    lambda2(other.lambda2),
+    elasticNet(other.elasticNet),
+    tolerance(other.tolerance),
+    betaPath(other.betaPath),
+    lambdaPath(other.lambdaPath),
+    activeSet(other.activeSet),
+    isActive(other.isActive),
+    ignoreSet(other.ignoreSet),
+    isIgnored(other.isIgnored)
+{
+  // Nothing to do here.
+}
+
+// Move constructor.
+LARS::LARS(LARS&& other) :
+    matGramInternal(std::move(other.matGramInternal)),
+    matGram(other.matGram),
```

Comment:
I don't think you need to do this. If you are setting the values inside a constructor. Same with lambda1 and lambda2.

Reply:
Actually there's a bit more trickiness to this than it looks like from the outset.  The `matGram` member is a pointer to either a locally-held Gram matrix (`matGramInternal`) *or* a pointer to an externally-provided Gram matrix, for which we're not responsible for the memory.

So, this code works correctly if `other.matGram` points to an externally-defined matrix.  (i.e., `other.matGram != &other.matGramInternal`)  But if `other.matGram == other.matGramInternal`, then we actually want to set `matGram` to point to this object's `matGramInternal`, instead of `other.matGram`, since we've just done a `std::move()` on the Gram matrix and `other.matGramInternal` is not empty.  I hope this makes sense; let me know if I can clarify. :+1:

Reply:
That makes sense thanks for clarifying.

Reply:
Hi @rcurtin, Thanks for the info. I agree, that makes sense. I will make the changes and push them shortly. Sorry it took a while to response, I had some deadlines that I had to meet today. Thanks a lot.